src/OmniSharp/Settings/FromBodyApplicationModelConvention.cs

```diff
@@ -15,8 +15,8 @@ public void Apply(ApplicationModel application)
                     foreach (var parameter in action.Parameters)
                     {
                         if (parameter.BindingInfo?.BindingSource != null ||
-                            parameter.Attributes.OfType<IBindingSourceMetadata>().Any() ||
-                            ValueProviderResult.CanConvertFromString(parameter.ParameterInfo.ParameterType))
+                            parameter.Attributes.OfType<IBindingSourceMetadata>().Any())
+                            //|| ValueProviderResult.CanConvertFromString(parameter.ParameterInfo.ParameterType))
```

Comment:
Looks like this is here to mimic WebAPI2's behavior? IMO it's best for APIs to be explicit. You could also go the route of `[FromBody]` by default and have anything else be explicit.